File: app/javascripts/models/payment.js
//= require "../model"

var Payment = new JS.Class(Model, {
  extend: {
    resource: 'payment',
    columns: ['id', 'transaction_id', 'form', 'amount'],
    belongs_to: ['transaction']
  },
  
  valid: function() {
    return true;
  }
});

File: app/models/payment.rb
class Payment < ActiveRecord::Base
  attr_readonly           :transaction_id
  validates_presence_of   :form, :amount
  validates_inclusion_of  :form, :in => %w(cash check credit_card gift_card store_credit), :message => "%{value} is not a valid payment type"
  
  belongs_to              :transaction
end